File: engine/schemas/table_design.py
from pydantic import BaseModel, Field

class TableDesignColumnRequest(BaseModel):
    name: str
    type: str
    nullable: bool = True
    default_value: str | None = None
    primary_key: bool = False
    auto_increment: bool = False
    comment: str | None = None


class TableDesignIndexRequest(BaseModel):
    name: str | None = None
    columns: list[str]
    unique: bool = False


class TableDesignDDLRequest(BaseModel):
    table_name: str
    table_comment: str | None = None
    engine: str = "InnoDB"
    charset: str = "utf8mb4"
    collation: str = "utf8mb4_0900_ai_ci"
    columns: list[TableDesignColumnRequest]
    indexes: list[TableDesignIndexRequest] = Field(default_factory=list)


class TableDesignExecuteRequest(BaseModel):
    datasource_id: str
    ddl: str
    confirm_token: str | None = None
    confirm_text: str | None = None


class TableDesignDraftSaveRequest(BaseModel):
    project_id: str
    draft_id: str | None = None
    table_name: str
    table_comment: str | None = None
    columns: list[TableDesignColumnRequest]
    indexes: list[TableDesignIndexRequest] = Field(default_factory=list)


class TableDesignAIRequest(BaseModel):
    prompt: str
    api_key: str | None = None
    api_base: str | None = None
    model_name: str | None = None


class TestDataGenerateRequest(BaseModel):
    datasource_id: str
    table_name: str
    row_count: int = 10
    language: str = "zh"
    confirm_token: str | None = None
    confirm_text: str | None = None

File: desktop/src/lib/api/tableDesign.ts
import { request } from "./client";
import type {
  DangerousOperationResult,
  DeleteResponse,
  TableDesignAiResponse,
  TableDesignColumn,
  TableDesignDraft,
  TableDesignExecutionResult,
  TableDesignIndex,
} from "./types";

export const tableDesignApi = {
  executeTableDesignDDL: (datasourceId: string, ddl: string, confirm?: { token: string; text: string }) =>
    request<DangerousOperationResult<TableDesignExecutionResult>>("/schema/design/execute-ddl", {
      method: "POST",
      body: JSON.stringify({
        datasource_id: datasourceId,
        ddl,
        confirm_token: confirm?.token,
        confirm_text: confirm?.text,
      }),
    }),

  generateTestData: (params: { datasource_id: string; table_name: string; row_count?: number; language?: string }, confirm?: { token: string; text: string }) =>
    request<DangerousOperationResult<{ success: boolean; tableName: string; insertedRows: number; latencyMs: number; message: string }>>("/schema/generate-test-data", {
      method: "POST",
      body: JSON.stringify({
        ...params,
        confirm_token: confirm?.token,
        confirm_text: confirm?.text,
      }),
    }),

  listTableDesignDrafts: (projectId: string) =>
    request<TableDesignDraft[]>(`/schema/design/drafts?project_id=${projectId}`),

  getTableDesignDraft: (draftId: string) =>
    request<TableDesignDraft>(`/schema/design/drafts/${draftId}`),

  saveTableDesignDraft: (req: {
    project_id: string;
    draft_id?: string;
    table_name: string;
    table_comment?: string;
    columns: TableDesignColumn[];
    indexes: TableDesignIndex[];
  }) =>
    request<TableDesignDraft>("/schema/design/drafts/save", {
      method: "POST",
      body: JSON.stringify(req),
    }),

  deleteTableDesignDraft: (draftId: string) =>
    request<DeleteResponse>(`/schema/design/drafts/${draftId}`, { method: "DELETE" }),

  generateTableDesignAi: (prompt: string, config?: { apiKey?: string; apiBase?: string; model?: string }) =>
    request<TableDesignAiResponse>("/schema/design/ai-generate", {
      method: "POST",
      body: JSON.stringify({
        prompt,
        api_key: config?.apiKey,
        api_base: config?.apiBase,
        model_name: config?.model,
      }),
    }),
};
